dashboard/app/models/script_level.rb

```diff
@@ -41,13 +38,30 @@ class ScriptLevel < ActiveRecord::Base
     progression
     target
     challenge
+    type
   )
 
+  NAMED_LEVEL = 'named_level'
+  ASSESSMENT = 'assessment'
+  BONUS = 'bonus'
+
   def script
     return Script.get_from_cache(script_id) if Script.should_cache?
     super
   end
 
+  def named_level
+    type == NAMED_LEVEL
+  end
+
+  def assessment
+    type == ASSESSMENT
+  end
+
+  def bonus
+    type == BONUS
+  end
```

Comment:
These are replicating the existing boolean column accessors that I'm removing.  In the future my plan is to update any code that does e.g. `script_level.bonus` with `script_level.type == bonus`.